src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/close_application_request_test.cc

```diff
@@ -0,0 +1,308 @@
+        .WillByDefault(ReturnRef(mock_connection_handler_));
+    ON_CALL(mock_connection_handler_, get_session_observer())
+        .WillByDefault(ReturnRef(mock_session_handler_));
+    ON_CALL(*mock_app_, app_id()).WillByDefault(Return(kConnectionKey));
+  }
+
+  MockAppPtr mock_app_;
+  MessageSharedPtr message_;
+  MessageSharedPtr response_message_;
+  NiceMock<components::connection_handler_test::MockConnectionHandler>
+      mock_connection_handler_;
+  components::protocol_handler_test::MockSessionObserver mock_session_handler_;
+  CloseApplicationCommandPtr command_;
+};
+
+TEST_F(CloseApplicationRequestTest,
+       CloseApplicationRequestSendsResponse_APPLICATION_NOT_REGISTERED) {
+  using namespace am;
+  auto message = CreateCloseAppMessage();
+
+  CloseApplicationCommandPtr command(
+      CreateCommand<sdl_rpc_plugin::commands::CloseApplicationRequest>(
+          message));
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(MockAppPtr()));
+
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).Times(0);
+
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(_, am::commands::Command::CommandSource::SOURCE_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&response_message_), Return(true)));
+  command->Run();
+
+  CheckResponseParameters(false,
+                          mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+}
+
+TEST_F(CloseApplicationRequestTest,
+       CloseApplicationRequestSendsResponse_IGNORED) {
+  using namespace am;
+  auto message = CreateCloseAppMessage();
+
+  CloseApplicationCommandPtr command(
+      CreateCommand<sdl_rpc_plugin::commands::CloseApplicationRequest>(
+          message_));
+  MockAppPtr mock_app(CreateMockApp());
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  EXPECT_CALL(*mock_app, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::eType::HMI_NONE));
+
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).Times(0);
+
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(_, am::commands::Command::CommandSource::SOURCE_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&response_message_), Return(true)));
+  command->Run();
+
+  CheckResponseParameters(false, mobile_apis::Result::IGNORED);
+}
+
+TEST_F(CloseApplicationRequestTest,
+       CloseApplicationRequestSendsResponse_INVALID_DATA) {
+  using namespace am;
+  CloseApplicationCommandPtr command(
+      CreateCommand<sdl_rpc_plugin::commands::CloseApplicationRequest>(
+          message_));
+
+  DeffaultRequestRunCheck(false);
+
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(_, am::commands::Command::CommandSource::SOURCE_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&response_message_), Return(true)));
+
+  command->Run();
+
+  CheckResponseParameters(false, mobile_apis::Result::INVALID_DATA);
+}
+
+TEST_F(CloseApplicationRequestTest,
+       CloseApplicationRequestSendsResponse_GENERIC_ERROR) {
```

Comment:
@IGapchuk please use naming pattern "WhatIsTested_TestScenario_ExpectedResult" pattern for your unit tests

Reply:
@AKalinich-Luxoft Renamed in 552836a13c762ae1f0ab5e7eec2c38c4f796528f.